src/algebra/group_power/basic.lean

```diff
@@ -377,6 +377,13 @@ end comm_group
 lemma zero_pow [monoid_with_zero R] : ∀ {n : ℕ}, 0 < n → (0 : R) ^ n = 0
 | (n+1) _ := zero_mul _
 
+lemma zero_pow_eq [monoid_with_zero R] {n : ℕ} : (0 : R)^n = if n = 0 then 1 else 0 :=
```

Comment:
```suggestion
lemma zero_pow_eq [monoid_with_zero R] (n : ℕ) : (0 : R)^n = if n = 0 then 1 else 0 :=
```
since there's nothing to infer `n` from in the other hypotheses, unlike the one above

Reply:
Is `pi.single 0 1 n` a more or less useful `RHS`? In theory, you could state this:
```suggestion
lemma zero_pow_eq [monoid_with_zero R] {n : ℕ} : (^) (0 : R) = pi.single 0 1 := funext $ sorry
```
which would be match more generally.

Reply:
Hmm, I think I'd prefer to leave that one.